src/comment_parser.rs

```diff
@@ -0,0 +1,379 @@
+//! Implements markup parsing for metamath theorem comments.
+//!
+//! The [set.mm](https://github.com/metamath/set.mm/) library contains many
+//! theorems with comments, and those comments contain markup in a special syntax
+//! implemented by [metamath.exe](https://github.com/metamath/metamath-exe).
+//! This module implements a SAX-style parser which yields events about the
+//! beginning and end of each syntax event.
+//!
+//! The [`CommentItem`] type does not contain any byte buffers directly;
+//! instead everything uses spans relative to the segment in which this comment
```

Comment:
`Span` is surely the most efficient type since no copy is involved whatsoever. Could we have an API which turns these Spans into `&'a str`, given a `StatementRef<'a>`?
I believe the whole segmenting shall be kept from the API.

Reply:
I believe the segment buffer is exposed in the API, even if segments aren't.